terraform/modules/app/main.tf

```diff
@@ -67,3 +67,13 @@ resource "aws_security_group" "service" {
   vpc_id      = var.vpc_id
   description = "${var.service_name} app ECS tasks"
 }
+
+resource "aws_security_group_rule" "statsd_from_apps_tcp" {
```

Comment:
Not necessarily a blocker for this PR, but I wonder if we should have a separate security group for "all the apps in the mesh", so we can represent the kinds of statsd / prometheus situations a bit more cleanly.

So the security group rule would be "any app in this security group can talk to statsd", rather than having a separate rule for every app.

Later on we're going to need to say "all apps can talk to prometheus", and probably other similar things, which would be easier if we had a single group that represented all the apps.

Reply:
That sounds reasonable to me. Perhaps we can do this when we create Prometheus?

I had an impression from working with govuk-aws that shared security groups (like `govuk_management_access`) were an anti pattern, but I'm not speaking from much experience. I wonder if this could lead to folks being tempted to use the "app_in_mesh" group inappropriately?

One thing to watch out for is that ECS Services are permitted a maximum of 5 security groups. This isn't a blocker for doing this, just something to be aware of given some apps would have 3 security groups if we went down this route.

Reply:
Ultimately we don't want to be passing security groups down into every module if we can help it. Having an "all services" group sounds like a good way to do that. You're right that we do have to use security groups sparingly but this is a pretty compelling reason. One for https://trello.com/c/QnM38plJ/31-plan-architecture-of-vpc-network-acls-and-security-groups

Reply:
(I'm also ok with this for now btw, as long as everyone knows it's not something to be copy/pasted!)

Reply:
Thanks both, this seems like a good approach so I'll fix it up in this commit.

Reply:
I've made this change